internal/isoeditor/rhcos.go

```diff
@@ -95,9 +106,48 @@ func formatRootfsServiceConfigFile(clusterProxyInfo *ClusterProxyInfo) (string,
 }
 
 func addFileToArchive(w *cpio.Writer, path string, content string, mode cpio.FileMode) error {
+	err := addDirToArchive(w, path)
+	if err != nil {
+		return err
+	}
+	hdr := &cpio.Header{
+		Name: path,
+		Mode: mode,
+		Size: int64(len(content)),
+	}
+	if err := w.WriteHeader(hdr); err != nil {
+		return err
+	}
+	if _, err := w.Write([]byte(content)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func addSymlinkToArchive(w *cpio.Writer, path string, target string) error {
+	err := addDirToArchive(w, path)
+	if err != nil {
+		return err
+	}
+	content := []byte(target)
+	hdr := &cpio.Header{
+		Name: path,
+		Mode: cpio.ModeSymlink | 0777,
+		Size: int64(len(content)),
+	}
+	if err := w.WriteHeader(hdr); err != nil {
+		return err
+	}
+	if _, err := w.Write(content); err != nil {
+		return err
+	}
+	return nil
+}
+
+func addDirToArchive(w *cpio.Writer, path string) error {
 	// add all the directories in path in the correct order, using dirsStack as FILO
 	dirsStack := []string{}
-	for dir := filepath.Dir(path); dir != "" && dir != "/"; dir = filepath.Dir(dir) {
+	for dir := filepath.Dir(path); dir != "" && dir != "/" && dir != "."; dir = filepath.Dir(dir) {
```

Comment:
Is this fixing a particular bug? If so, it would be good to call out what was going on in a separate commit.

Reply:
While trying to figure out how to write symlinks I added a `a -> b` symlink and this code got into an endless loop. This fixes that. I will move that to a separate patch.